core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -2482,6 +2485,38 @@ private static void assertVisibleCount(InternalEngine engine, int numDocs, boole
         }
     }
 
+    public void testTranslogCleanUpPostCommitCrash() throws Exception {
+        try (Store store = createStore()) {
+            AtomicBoolean throwErrorOnCommit = new AtomicBoolean();
+            final Path translogPath = createTempDir();
+            try (InternalEngine engine = new InternalEngine(config(defaultSettings, store, translogPath, newMergePolicy(), null, null)) {
+                @Override
+                protected void commitIndexWriter(IndexWriter writer, Translog translog, String syncId) throws IOException {
```

Comment:
@s1monw I tried this is in many variants and non of them was good. The Failure as states fails too early (before the `segmetns_N` file is written). I tried many other variants but none of them allow all of the commit logic to complete without triggering any failure handling in the IndexWriter which also means later on that the new commits files are cleaned by a rollback we do when we fail the engine. Even if we do find a way to do this, I think it will be way too brittle and tend to break with changes in Lucene. Bottom line, I prefer to keep the current solution.